### PR TITLE
Quality: code cleaning

### PR DESCRIPTION
**Pull request by EtienneLt**
**Please check if the PR fulfills these requirements**

- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem?**
No

**What kind of change does this PR introduce?**
Cleaning some code:
- Use Java14 switch syntax ([JEP-361](https://openjdk.org/jeps/361)), especially `return switch` syntax.
- Replace some `Stream.collect(Collector.toList())` with `Stream.toList()` (when immutable lists can be returned without impact for callers).
- Use `JsonParser.currentName()` instead of deprecated method `JsonParser.getCurrentName()`.
- Replace `Files.write(file, string.getBytes(StandardCharsets.UTF_8))` with `Files.writeString(file, string)`.
- Replace some `Arrays.asList(...)` with `List.of(...)`.
- Replace `assertTrue(obj instanceof Foo)` with `assertInstanceOf(Foo.class, obj)`.
- A few other simplifications.

**Does this PR introduce a breaking change or deprecate an API?**
- [ ] Yes
- [x] No

### CODE REVIEW

**Review by geofjamg — COMMENTED** [review]
Be careful, each time you replace a collec(Collectors.toList())) by a toList() and that it is returned from an api you break potentialy the api because list is now immutable. And using code outside of powsybl core might failed at runtime.

**Comment by EtienneLt** [comment]
> Be careful, each time you replace a collec(Collectors.toList())) by a toList() and that it is returned from an api you break potentialy the api because list is now immutable. And using code outside of powsybl core might failed at runtime.

ok I can revert if needed, I left many Collectors.toList()

**Comment by geofjamg** [comment]
> > Be careful, each time you replace a collec(Collectors.toList())) by a toList() and that it is returned from an api you break potentialy the api because list is now immutable. And using code outside of powsybl core might failed at runtime.
> 
> ok I can revert if needed, I left many Collectors.toList()

I just say that each time a list is exposed to an API we have to think and decide if we want to expose a mutable or immutable list and report if we change from actual code as a breaking change and write it in the migration guide

File: time-series/time-series-api/src/main/java/com/powsybl/timeseries/DataChunk.java [hunk 1]
```diff
@@ -123,15 +123,13 @@ A getChunk2() {
      * Append the chunk with the one given in argument, and return the result. "This" dataChunk and the one in argument remain unchanged.
      * The two chunks have to be successive, i.e : this.getOffset() + this.length() = otherChunk.getOffset()
      * @param otherChunk : the chunk to append with this object. It has to be the same implementation as this object.
```
Comment (rolnico) [inline time-series/time-series-api/src/main/java/com/powsybl/timeseries/DataChunk.java:125]: delete `:`

**Review by rolnico — CHANGES_REQUESTED** [review]

**Review by olperr1 — CHANGES_REQUESTED** [review]

File: powerfactory/powerfactory-converter/src/main/java/com/powsybl/powerfactory/converter/LineConverter.java [hunk 2]
```diff
@@ -75,7 +76,9 @@ private void createFromElmLneFromElmTow(DataObject elmTow, DataObject elmLne) {
 
         NodeRef end1 = nodeRefs.get(0);
         NodeRef end2 = nodeRefs.get(1);
-
+        if (lineModel.isEmpty()) {
+            throw new PowsyblException("line Model cant be null to create a Line");
+        }
```
Comment (olperr1) [inline powerfactory/powerfactory-converter/src/main/java/com/powsybl/powerfactory/converter/LineConverter.java:81]: Since `lineModel` can't be empty, Sonar raise an issue on this test.

I think the good way is to change the return type of `LineModel.createFromElmTow(...)`. There is no need for it to return an `Optional` and it can directly return the `LineModel`.

File: powerfactory/powerfactory-converter/src/main/java/com/powsybl/powerfactory/converter/TransformerConverter.java [hunk 3]
```diff
@@ -178,8 +177,8 @@ void createThreeWindings(DataObject elmTr3) {
         Optional<PhaseAngleClockModel> pac3 = pac3WModel.getEnd(windingTypeEnds.get(2));
         if (pac2.isPresent() || pac3.isPresent()) {
             t3wt.newExtension(ThreeWindingsTransformerPhaseAngleClockAdder.class)
-                .withPhaseAngleClockLeg2(pac2.isPresent() ? pac2.get().pac : 0)
-                .withPhaseAngleClockLeg3(pac3.isPresent() ? pac3.get().pac : 0).add();
+                .withPhaseAngleClockLeg2(pac2.map(phaseAngleClockModel -> phaseAngleClockModel.pac).orElse(0))
+                .withPhaseAngleClockLeg3(pac3.map(angleClockModel -> angleClockModel.pac).orElse(0)).add();
```
Comment (olperr1) [inline powerfactory/powerfactory-converter/src/main/java/com/powsybl/powerfactory/converter/TransformerConverter.java:181]: Can you use the same variable name for both lambda? (And maybe a shorter one, as `model`?)

File: psse/psse-model/src/main/java/com/powsybl/psse/model/pf/PsseValidation.java [hunk 4]
```diff
@@ -210,19 +205,18 @@ private void validateNonTransformerBranches(List<PsseNonTransformerBranch> nonTr
 
     private void validateTransformers(List<PsseTransformer> transformers, Map<Integer, List<Integer>> buses) {
         List<PsseTransformer> twoWinddingsTransformers = transformers.parallelStream()
```
Comment (olperr1) [inline psse/psse-model/src/main/java/com/powsybl/psse/model/pf/PsseValidation.java:207]: Can you fix this variable name?

File: psse/psse-model/src/main/java/com/powsybl/psse/model/pf/PsseValidation.java [hunk 5]
```diff
@@ -210,19 +205,18 @@ private void validateNonTransformerBranches(List<PsseNonTransformerBranch> nonTr
...
-            .filter(transformer -> transformer.getK() == 0).collect(Collectors.toList());
+            .filter(transformer -> transformer.getK() == 0).toList();
         validateTwoWindingsTransformers(twoWinddingsTransformers, buses);
 
         List<PsseTransformer> threeWinddingsTransformers = transformers.parallelStream()
```
Comment (olperr1) [inline psse/psse-model/src/main/java/com/powsybl/psse/model/pf/PsseValidation.java:211]: Idem here

File: psse/psse-model/src/main/java/com/powsybl/psse/model/pf/io/MultiTerminalDcTransmissionLineData.java [hunk 6]
```diff
@@ -249,9 +248,9 @@ private static List<PsseMultiTerminalDcTransmissionLine> convertToMultiTerminalL
...
-                List<PsseMultiTerminalDcConverter> converterList = converterxList.stream().filter(c -> c.getName().equals(main.getName())).map(PsseMultiTerminalDcConverterx::getConverter).collect(Collectors.toList());
-                List<PsseMultiTerminalDcBus> busList = busxList.stream().filter(c -> c.getName().equals(main.getName())).map(PsseMultiTerminalDcBusx::getBus).collect(Collectors.toList());
-                List<PsseMultiTerminalDcLink> linkList = linkxList.stream().filter(c -> c.getName().equals(main.getName())).map(PsseMultiTerminalDcLinkx::getLink).collect(Collectors.toList());
+                List<PsseMultiTerminalDcConverter> converterList = converterxList.stream().filter(c -> c.getName().equals(main.getName())).map(PsseMultiTerminalDcConverterx::getConverter).toList();
+                List<PsseMultiTerminalDcBus> busList = busxList.stream().filter(c -> c.getName().equals(main.getName())).map(PsseMultiTerminalDcBusx::getBus).toList();
+                List<PsseMultiTerminalDcLink> linkList = linkxList.stream().filter(c -> c.getName().equals(main.getName())).map(PsseMultiTerminalDcLinkx::getLink).toList();
```
Comment (olperr1) [inline psse/psse-model/src/main/java/com/powsybl/psse/model/pf/io/MultiTerminalDcTransmissionLineData.java:253]: Could you revert these 3 changes?
For instance `converterList` may be modified in `MultiTerminalDcTransmissionLineData.addNextN(...)`.

File: triple-store/triple-store-api/src/main/java/com/powsybl/triplestore/api/PropertyBag.java [hunk 7]
```diff
@@ -154,9 +154,9 @@ private String tabulate(String title, BiFunction<PropertyBag, String, String> ge
             String format = String.format("%%-%ds", lenPad);
 
             // Performance : avoid using concat() -> use a StringBuilder instead.
-            return new StringBuilder(title).append(lineSeparator).append(propertyNames.stream()
-                    .map(n -> new StringBuilder(INDENTATION).append(String.format(format, n)).append(" : ").append(getValue.apply(this, n)).toString())
-                    .collect(Collectors.joining(lineSeparator))).toString();
+            return title + lineSeparator + propertyNames.stream()
+                .map(n -> INDENTATION + String.format(format, n) + " : " + getValue.apply(this, n))
+                .collect(Collectors.joining(lineSeparator));
```
Comment (olperr1) [inline triple-store/triple-store-api/src/main/java/com/powsybl/triplestore/api/PropertyBag.java:159]: This change should be reverted.
If you look at the comment, using the `StringBuilder` instead of concats (or `+`) was done for performance purposes.

File: triple-store/triple-store-api/src/main/java/com/powsybl/triplestore/api/PropertyBags.java [hunk 8]
```diff
@@ -115,14 +115,12 @@ private String tabulate(BiFunction<PropertyBag, String, String> getValue) {
         String columnSeparator = " \t ";
         String lineSeparator = System.lineSeparator();
 
-        StringBuilder s = new StringBuilder(size() * 80);
```
Comment (olperr1) [inline triple-store/triple-store-api/src/main/java/com/powsybl/triplestore/api/PropertyBags.java:118]: I think it would be wiser to revert this change to.
It may also impact the performances (same as previous comment).

**Review by olperr1 — CHANGES_REQUESTED** [review]

File: iidm/iidm-impl/src/test/java/com/powsybl/iidm/network/impl/BranchTest.java [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+        Assertions.assertEquals(500, branch2.getNullableCurrentLimits2().getPermanentLimit());
+        Assertions.assertEquals(0, branch2.getNullableCurrentLimits2().getTemporaryLimits().size());
+
+        Assertions.assertNotNull(branch2.getCurrentLimits(TwoSides.ONE));
+        Assertions.assertNotNull(branch2.getCurrentLimits(TwoSides.TWO));
+        Assertions.assertNotNull(branch2.getNullableCurrentLimits(TwoSides.ONE));
+        Assertions.assertNotNull(branch2.getNullableCurrentLimits(TwoSides.TWO));
+
+        Assertions.assertNotNull(branch2.getNullableLimits(LimitType.CURRENT, TwoSides.ONE));
+        Assertions.assertNotNull(branch2.getNullableLimits(LimitType.CURRENT, TwoSides.TWO));
```
Comment (olperr1) [inline iidm/iidm-impl/src/test/java/com/powsybl/iidm/network/impl/BranchTest.java:89]: Could you also add a test on the permanent limit value for each non-null item.
In the current state, nothing verify that the returned values are the expected one (for instance, side 1 and side 2 could be mixed here).

File: iidm/iidm-impl/src/test/java/com/powsybl/iidm/network/impl/BranchTest.java [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+        Assertions.assertNotNull(branch1.getNullableActivePowerLimits(TwoSides.TWO));
+        Assertions.assertNotNull(branch1.getApparentPowerLimits(TwoSides.ONE));
+        Assertions.assertNotNull(branch1.getApparentPowerLimits(TwoSides.TWO));
+        Assertions.assertNotNull(branch1.getNullableApparentPowerLimits(TwoSides.ONE));
+        Assertions.assertNotNull(branch1.getNullableApparentPowerLimits(TwoSides.TWO));
+
+        Assertions.assertNotNull(branch1.getNullableLimits(LimitType.ACTIVE_POWER, TwoSides.ONE));
+        Assertions.assertNotNull(branch1.getNullableLimits(LimitType.ACTIVE_POWER, TwoSides.TWO));
+        Assertions.assertNotNull(branch1.getNullableLimits(LimitType.APPARENT_POWER, TwoSides.ONE));
+        Assertions.assertNotNull(branch1.getNullableLimits(LimitType.APPARENT_POWER, TwoSides.TWO));
```
Comment (olperr1) [inline iidm/iidm-impl/src/test/java/com/powsybl/iidm/network/impl/BranchTest.java:60]: Same here.

File: iidm/iidm-impl/src/test/java/com/powsybl/iidm/network/impl/BranchTest.java (hunk 9 above)
Comment (EtienneLt) [reply iidm/iidm-impl/src/test/java/com/powsybl/iidm/network/impl/BranchTest.java:89]: ok

**Review by olperr1 — CHANGES_REQUESTED** [review]

File: cgmes/cgmes-conversion/src/test/java/com/powsybl/cgmes/conversion/test/network/compare/Comparison.java [hunk 11]
```diff
@@ -524,16 +524,16 @@ private void compareGeneratorReactiveCapabilityCurve(
...
-                .collect(Collectors.toList());
+                .toList();
         compare("reactiveCapabilityCurve.size", e.size(), a.size());
         for (int k = 0; k < e.size(); k++) {
             Point pe = e.get(k);
             Point pa = a.get(k);
             compare("reactiveCapabilityCurvePoint.p", pe.getP(), pa.getP());
             compare("reactiveCapabilityCurvePoint.minQ", pe.getMinQ(), pa.getMinQ());
-            compare("reactiveCapabilityCurvePoint.maxQ", pe.getMaxQ(), pa.getMaxQ());
+            compare("reactiveCapabilityCurvebPoint.maxQ", pe.getMaxQ(), pa.getMaxQ());
```
[the remark this answers is not in the document]
Comment (EtienneLt) [reply cgmes/cgmes-conversion/src/test/java/com/powsybl/cgmes/conversion/test/network/compare/Comparison.java:536]: yes I fixed it

[the remark this answers is not in the document]
Comment (EtienneLt) [reply cgmes/cgmes-conversion/src/test/java/com/powsybl/cgmes/conversion/test/network/compare/Comparison.java]: fixed

File: powerfactory/powerfactory-model/src/main/java/com/powsybl/powerfactory/model/StudyCase.java [hunk 12]
```diff
@@ -62,30 +61,31 @@ static StudyCase parseJson(JsonParser parser) {
...
+            }
+            case "objects" -> {
+                JsonUtil.parseObjectArray(parser, obj -> { }, parser2 -> DataObject.parseJson(parser2, context.index, context.scheme));
+                yield true;
+            }
+            case "elmNets" -> {
+                context.elmNets = JsonUtil.parseLongArray(parser).stream()
+                    .map(id -> context.index.getDataObjectById(id)
+                        .orElseThrow(() -> new PowerFactoryException("ElmNet object " + id + " not found")))
+                    .toList();
```
Comment (olperr1) [inline powerfactory/powerfactory-model/src/main/java/com/powsybl/powerfactory/model/StudyCase.java:85]: I think it would be safer to revert to `.collect(Collectors.toList())` here.

File: psse/psse-model/src/main/java/com/powsybl/psse/model/pf/io/MultiTerminalDcTransmissionLineData.java [hunk 13]
```diff
@@ -221,7 +221,7 @@ public void write(List<PsseMultiTerminalDcTransmissionLine> multiTerminalList, C
             if (outputStream != null) {
                 throw new PsseException("Unexpected outputStream. Should be null");
             }
-            List<PsseMultiTerminalDcMain> mainList = multiTerminalList.stream().map(PsseMultiTerminalDcTransmissionLine::getMain).collect(Collectors.toList());
+            List<PsseMultiTerminalDcMain> mainList = multiTerminalList.stream().map(PsseMultiTerminalDcTransmissionLine::getMain).toList();
```
Comment (olperr1) [inline psse/psse-model/src/main/java/com/powsybl/psse/model/pf/io/MultiTerminalDcTransmissionLineData.java:224]: Here too.

**Review by olperr1 — CHANGES_REQUESTED** [review]

**Review by olperr1 — APPROVED** [review]
